lib/data_explorer_query_group_bookmarkable.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class DataExplorerQueryGroupBookmarkable < BaseBookmarkable
+  def self.model
+    DataExplorer::QueryGroup
+  end
+
+  def self.serializer
+    UserDataExplorerQueryGroupBookmarkSerializer
+  end
+
+  def self.preload_associations
+    [:data_explorer_queries, :groups]
+  end
+
+  def self.list_query(user, guardian)
+    group_ids = user.groups.pluck(:id) 
+    return if group_ids.empty?
+    user.bookmarks_of_type("DataExplorer::QueryGroup")
+    .joins("INNER JOIN data_explorer_query_groups ON data_explorer_query_groups.id = bookmarks.bookmarkable_id
+          AND bookmarks.bookmarkable_type = 'DataExplorer::QueryGroup'")
+      .joins("LEFT JOIN data_explorer_queries ON data_explorer_queries.id = data_explorer_query_groups.query_id")
+      .joins("LEFT JOIN groups ON groups.id = data_explorer_query_groups.group_id")
+      .where("data_explorer_query_groups.group_id IN (?)", group_ids)
```

Comment:
I think we only need to conditionally add this if user is not staff -- staff can see every group. Alternatively, we can just not allow admins to bookmark these queries within a group if they are not part of that group. Thoughts @tgxworld ?

Reply:
Staff can see every group, but it looks like unless the Staff is a member of the group, they can't see the Reports page within the Group. So I think it follows that they shouldn't see the bookmarks if they're not a member of the group either.

For Admin, though, they *can* see the Reports page of a Group they're not a member of, so I can add a condition to `list_query` to consider this. Unless we decide admins shouldn't be able to bookmark from these groups? (They're admins, though. Aren't they usually allowed to do everything while ignoring permissions?)

Reply:
Ah sorry I meant to say `admin` not `staff`, my bad.

Reply:
So yes we need the condition I mentioned but for `admin` users only.

Reply:
Alright then, will apply the condition for admins

Reply:
Done!